org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalReplacementProvider.java

```diff
@@ -189,6 +190,10 @@ public void updateReplacement(CompletionProposal proposal, CompletionItem item,
 		} else {
 			item.setInsertTextFormat(InsertTextFormat.PlainText);
 		}
+		if (!JavaLanguageServerPlugin.getPreferencesManager().getClientPreferences().isCompletionListItemDefaultsSupport("insertTextMode") &&
```

Comment:
I think we should only do this kind of check when it's in `textDocument/completion` request. And always set the insertTextMode when it's `completionItem/resolve`.

You can get the state by checking the member `isResolvingRequest`.